xdem/spatialstats.py

```diff
@@ -1394,6 +1802,142 @@ def neff_hugonnet_approx(coords: np.ndarray, errors: np.ndarray, params_variogra
+            y = rasterize_resolution * np.arange(0, mask.shape[1])
+            coords = np.array(np.meshgrid(y, x))
+            coords_on_mask = coords[:, mask].T
+
+        elif isinstance(rasterize_resolution, Raster):
+
+            # With a Raster we can get the coordinates directly
+            mask = V.create_mask(rst=rasterize_resolution).squeeze()
+            coords = np.array(rasterize_resolution.coords())
+            coords_on_mask = coords[:, mask].T
+
+        else:
+            raise ValueError('The rasterize resolution must be a float, integer or Raster subclass.')
+
+        # Here we don't care about heteroscedasticity, so all errors are standardized to one
+        errors_on_mask = np.ones(len(coords_on_mask))
+
+        neff = neff_hugonnet_approx(coords=coords_on_mask, errors=errors_on_mask,
+                                    params_variogram_model=params_variogram_model, **kwargs)
+
+    else:
+        raise ValueError('Area must be a float, integer, Vector subclass or geopandas dataframe.')
+
+    return neff
+
+def spatial_error_propagation(areas: list[float | VectorType | gpd.GeoDataFrame],
+                              errors: RasterType,
+                              params_variogram_model: pd.Dataframe,
+                              **kwargs) -> list[float]:
+    """
+    Spatial propagation of elevation errors to an area using the estimated heteroscedasticity and spatial correlations.
+
+    This function is based on the `number_effective_samples` function to estimate uncorrelated samples. If given a
+    vector area, it uses Equation 18 of Hugonnet et al. (2022), https://doi.org/10.1109/jstars.2022.3188922. If given
+    a numeric area, it uses a generalization of Rolstad et al. (2009), http://dx.doi.org/10.3189/002214309789470950.
+
+    The standard error SE (1-sigma) is then computed as SE = mean(SD) / Neff, where mean(SD) is the mean of errors in
+    the area of interest which accounts for heteroscedasticity, and Neff is the number of effective samples.
+
+    :param areas: Area of interest either as a numeric value of surface in the same unit as the variogram ranges (will
+        assume a circular shape), or as a vector (shapefile) of the area.
+    :param errors: Errors from heteroscedasticity estimation and modelling, as an array or Raster.
+    :param params_variogram_model: Dataframe of variogram models to sum with three to four columns, "model" for the model types
+        (e.g., ["spherical", "matern"]), "range" for the correlation ranges (e.g., [2, 100]), "psill" for the partial
+        sills (e.g., [0.8, 0.2]) and "smooth" for the smoothness parameter if it exists for this model (e.g.,
+        [None, 0.2]).
+    :param kwargs: Keyword argument to pass to the `neff_hugonnet_approx` function.
+
+    :return: List of standard errors (1-sigma) for the input areas
+    """
+
+    standard_errors = []
+    errors_arr = get_array_and_mask(errors)[0]
+    for area in areas:
```

Comment:
It would be great to take into account the case of a Vector or GeoDataFrame with several features and loop through each feature.

Reply:
Yes but not too urgent, because there's nothing that increases speed significantly by accounting for the loop within the function. And it would render parallelization delicate depending on the type of input (single Raster? VRT? etc).
Right now, a for loop as in the gallery example is sufficient.